closed/make/Javadoc.gmk

```diff
@@ -0,0 +1,475 @@
+# to be classes to be for the pages created in this build.
+JAVASE_BASE_URL := ../../$(JDKJRE2COREAPI)
+MGMT_BASE_URL := $(OPENJ9_MGMT_DOCDIR)
+OPENJ9_MGMT_SRCPATH := $(OPENJ9_DOCS_SRCDIR)$(PATH_SEP)$(TEMP_SUN_MGMT_PKGDIR)
+
+################################################################################
+# The javadoc tool only allows for a single setting for headers and footers
+# per invocation. When generating javadoc for the entire build, the pages for
+# the openj9 packages and classes therefore contain the openjdk page footer.
+#
+# The sed command below replaces the footer in the openj9 pages with the correct
+# values.
+#
+# '.' need escaping in the sed regex. If more special characters are introduced
+# in the openjdk JAVADOC_BOTTOM in future these would need escaping also.
+#
+# '&' needs escaping in sed replacement text.
+#
+# {@docroot} is expanded by javadoc when the html files are created, so the matching
+# needs to cater for anything {@docroot} may have been substituted by - '../../../..'
+# or '../..' or '..' etc.
+#
+# When pages are being generated solely for the openj9 classes, the correct
+# footer is specified directly in the javadoc command.
+
+################################################################################
+# \(\.\./\)*\(\.\.\)* means zero  or more '../' strings followed by
+# zero or more '..' strings.
+JAVADOC_DOCROOT := \(\.\./\)*\(\.\.\)*
+MGMT_BOTTOM_REGEX = $(subst {@docroot},$(JAVADOC_DOCROOT),$(subst .,\.,$(MGMT_BOTTOM)))
+OPENJ9_SUBST_BOTTOM = $(subst &,\&,$(OPENJ9_JAVADOC_BOTTOM))
+OPENJ9_MGMT_SUBST_COMMAND = sed -i 's|$(MGMT_BOTTOM_REGEX)|$(OPENJ9_SUBST_BOTTOM)|g'
+
+openj9-mgmtdocs: openj9-mgmtdocs-restore
+
+# Run javadoc if the package-list file is out of date or missing
+# The openjdk build already uses index.html as its marker, so this
+# uses the package-list file instead.
+# The dependency on the openjdk mgmtdocs ensures this javadoc generation
+# will replace the openjdk javadoc generation.
+$(OPENJ9_MGMT_PKGLIST_FILE): mgmtdocs openj9-mgmtdocs-prepare $(OPENJ9_MGMT_OPTIONS_FILE) $(OPENJ9_MGMT_PACKAGES_FILE)
+	echo Running $@
+	$(prep-javadoc)
+	$(call JavadocSummary,$(OPENJ9_MGMT_OPTIONS_FILE),$(OPENJ9_MGMT_PACKAGES_FILE))
+	$(JAVADOC_CMD) -d $(@D) @$(OPENJ9_MGMT_OPTIONS_FILE) @$(OPENJ9_MGMT_PACKAGES_FILE) 1>&2
+
+$(OPENJ9_MGMT_OPTIONS_FILE):
+	echo Running $@
+	$(prep-target)
+	@($(call OptionOnly,$(COMMON_JAVADOCFLAGS)) ; \
+	  $(call OptionPair,-sourcepath,$(OPENJ9_MGMT_SRCPATH)) ; \
+	  $(call OptionPair,-encoding,ascii) ; \
+	  $(call OptionPair,-doctitle,$(MGMT_DOCTITLE)) ; \
+	  $(call OptionPair,-windowtitle,$(MGMT_WINDOWTITLE) $(DRAFT_WINTITLE)); \
+	  $(call OptionPair,-header,$(MGMT_HEADER)$(DRAFT_HEADER)) ; \
+	  $(call OptionPair,-bottom,$(MGMT_BOTTOM)$(DRAFT_BOTTOM)) ; \
+	  $(call OptionTrip,-linkoffline,$(JAVASE_BASE_URL),$(COREAPI_DOCSDIR)/); \
+	  $(call OptionTrip,-linkoffline,$(MGMT_BASE_URL),$(MGMT_DOCDIR)/); \
```

Comment:
Inconsistent spacing: missing space before `;` to match lines above.

Reply:
Fixed